version: '3'

services:
  db:
    image: postgres:14.1-alpine
    container_name: postgres_container
    restart: always
    environment:
      POSTGRES_USER: ${POSTGRES_USER:-postgres}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD:-123}
      PGDATA: /var/lib/postgresql/data
    volumes:
       - postgres_data:/var/lib/postgresql/data
       - ./backup.sql:/docker-entrypoint-initdb.d/backup.sql
    ports:
      - "5555:5432"
    networks:
      - postgres
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
      interval: 1s
      timeout: 5s
      retries: 10

  pgadmin:
    container_name: pgadmin_container
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: ${PGADMIN_DEFAULT_EMAIL:-ale88andr@gmail.com}
      PGADMIN_DEFAULT_PASSWORD: ${PGADMIN_DEFAULT_PASSWORD:-123}
      PGADMIN_CONFIG_SERVER_MODE: 'False'
    volumes:
       - pgadmin_data:/var/lib/pgadmin

    ports:
      - "${PGADMIN_PORT:-5050}:80"
    networks:
      - postgres
    restart: unless-stopped

  cryman:
    container_name: cryman_container
    build: .

networks:
  postgres:
    driver: bridge

volumes:
  postgres_data:
  pgadmin_data: